### PR TITLE
[13.0] [MIG] website_sale_product_minimal_price: Migration to v13 

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT24175

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Check Travis and name 2 reviewers

**Review by HaraldPanten — APPROVED** [review]
Functional review 👍  Just check travis

**Comment by CarlosRoca13** [comment]
@sergio-teruel @Tardo please review it

File: website_sale_product_minimal_price/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Copyright 2019 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Website Sale Product Minimal Price",
+    "summary": "Display minimal price for products that has variants",
+    "version": "13.0.1.1.0",
```
Comment (Tardo) [inline website_sale_product_minimal_price/__manifest__.py]:
```suggestion
    "version": "13.0.1.0.0",
```

File: website_sale_product_minimal_price/controllers/main.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        methods=["POST"],
+        website=True,
+    )
+    def get_combination_info_minimal_price(self, product_template_ids, **kw):
+        """Special route to use website logic in get_combination_info override.
+        This route is called in JS by appending _website to the base route.
+        """
+
+        res = []
+        templates = request.env["product.template"].sudo().browse(product_template_ids)
```
Comment (Tardo) [inline website_sale_product_minimal_price/controllers/main.py:24]: This use the correct pricelist? You can see: https://github.com/odoo/odoo/blob/517a13b967a07fe1e71b9c5308d185c4e45f1a2d/addons/website_sale/controllers/main.py#L239

File: website_sale_product_minimal_price/static/src/js/website_sale_product_minimal_price.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
+odoo.define("website_sale_product_minimal_price.shop_min_price", function(require) {
+    "use strict";
+
+    var publicWidget = require("web.public.widget");
```
Comment (Tardo) [inline website_sale_product_minimal_price/static/src/js/website_sale_product_minimal_price.js]:
```suggestion
    const publicWidget = require("web.public.widget");
```

File: website_sale_product_minimal_price/static/src/js/website_sale_product_minimal_price.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    var core = require("web.core");
```
Comment (Tardo) [inline website_sale_product_minimal_price/static/src/js/website_sale_product_minimal_price.js]:
```suggestion
    const core = require("web.core");
```

File: website_sale_product_minimal_price/static/src/js/website_sale_product_minimal_price.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            "/website_sale_product_minimal_price/static/src/xml/website_sale_product_minimal_price.xml",
+        ],
+
+        start: function() {
+            return $.when.apply($, [
+                this._super.apply(this, arguments),
+                this.render_price(),
+            ]);
+        },
+        render_price: function() {
```
Comment (Tardo) [inline website_sale_product_minimal_price/static/src/js/website_sale_product_minimal_price.js]: To have a more readable source use 'await/async':
```suggestion
        render_price: async function() {
```

File: website_sale_product_minimal_price/static/src/js/website_sale_product_minimal_price.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+            const $products = $(".o_wsale_product_grid_wrapper");
+            const product_dic = {};
+            $products.each(function() {
+                product_dic[this.querySelector("a img").src.split("/")[6]] = this;
+            });
+            const product_ids = Object.keys(product_dic).map(Number);
+            return this._rpc({
```
Comment (Tardo) [inline website_sale_product_minimal_price/static/src/js/website_sale_product_minimal_price.js]:
```suggestion
            const product_min_price = await this._rpc({
```

File: website_sale_product_minimal_price/static/src/js/website_sale_product_minimal_price.js [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+                route: "/sale/get_combination_info_minimal_price/",
+                params: {product_template_ids: product_ids},
+            }).then(products_min_price => {
```
Comment (Tardo) [inline website_sale_product_minimal_price/static/src/js/website_sale_product_minimal_price.js]:
```suggestion
            });
```

File: website_sale_product_minimal_price/static/src/js/website_sale_product_minimal_price.js [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+                                        "website_sale_product_minimal_price.product_minimal_price",
+                                        {
+                                            price: this.widgetMonetary(product.price),
+                                        }
+                                    )
+                                ).get(0)
+                            );
+                    }
+                }
+            });
```
Comment (Tardo) [inline website_sale_product_minimal_price/static/src/js/website_sale_product_minimal_price.js:30]:
```suggestion
            }
            return products_min_price;
```

File: website_sale_product_minimal_price/static/src/js/website_sale_product_minimal_price.js [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+        selector: "#products_grid",
+        xmlDependencies: [
+            "/website_sale_product_minimal_price/static/src/xml/website_sale_product_minimal_price.xml",
+        ],
+
+        start: function() {
+            return $.when.apply($, [
+                this._super.apply(this, arguments),
+                this.render_price(),
+            ]);
```
Comment (Tardo) [inline website_sale_product_minimal_price/static/src/js/website_sale_product_minimal_price.js]: I think that here you can use ```Promise.all```
```suggestion
            return Promise.all([
                this._super.apply(this, arguments),
                this.render_price(),
            ]);
```

**Review by Tardo — CHANGES_REQUESTED** [review]
I think that you can make some changes to use more es6

File: website_sale_product_minimal_price/controllers/main.py (hunk 2 above)
Comment (CarlosRoca13) [reply website_sale_product_minimal_price/controllers/main.py:24]: With this, I take the product templates with the id that we took on the js.

Later, I catch the atribute with minor price by product template on this code: 

`cheaper_variant = template.product_variant_ids.sorted(
        key=lambda p: p._get_combination_info_variant().get("price")
 )[:1]`

File: .vscode/settings.json [hunk 10]
```diff
@@ -0,0 +1,3 @@
+{
+  "restructuredtext.confPath": ""
+}
```
Comment (sergio-teruel) [inline .vscode/settings.json]: Remove this file or ignore it

File: website_sale_product_minimal_price/static/src/js/website_sale_product_minimal_price.js [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+        },
+        render_price: async function() {
+            const $products = $(".o_wsale_product_grid_wrapper");
+            const product_dic = {};
+            $products.each(function() {
+                product_dic[this.querySelector("a img").src.split("/")[6]] = this;
+            });
+            const product_ids = Object.keys(product_dic).map(Number);
+            const products_min_price = await this._rpc({
```
Comment (sergio-teruel) [inline website_sale_product_minimal_price/static/src/js/website_sale_product_minimal_price.js:27]: Please @CarlosRoca13 as @Tardo comment, make sure that pricelist is in the context for get the correct price.

File: website_sale_product_minimal_price/controllers/main.py (hunk 2 above)
Comment (sergio-teruel) [reply website_sale_product_minimal_price/controllers/main.py:24]: Yes!!! review this

**Review by sergio-teruel — CHANGES_REQUESTED** [review]

**Review by sergio-teruel — APPROVED** [review]

**Comment by sergio-teruel** [comment]
Waiting travis

File: website_sale_product_minimal_price/controllers/main.py (hunk 2 above)
Comment (CarlosRoca13) [reply website_sale_product_minimal_price/controllers/main.py:24]: This is working. I test it with a custom pricelist

**Comment by CarlosRoca13** [comment]
@Tardo, can you review it now?

File: website_sale_product_minimal_price/static/src/js/website_sale_product_minimal_price.js [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+                route: "/sale/get_combination_info_minimal_price/",
+                params: {product_template_ids: product_ids},
+            });
+
+            for (const product of products_min_price) {
+                if (product.distinct_prices) {
```
Comment (Tardo) [inline website_sale_product_minimal_price/static/src/js/website_sale_product_minimal_price.js]: Changing this for something like:
```js
if (!product.distinct_prices) {
    continue;
}
```
you can save a level of indentation.

**Review by Tardo — APPROVED** [review]
:+1: LGTM. Only non-blocking comments.

**Comment by CarlosRoca13** [comment]
@Tardo change done!

**Pull request merged**